hazelcast-client/src/test/java/com/hazelcast/client/ClientConnectionTest.java

```diff
@@ -98,4 +105,40 @@ public void testMemberConnectionOrder() {
         Collection<Client> connectedClients2 = server2.getClientService().getConnectedClients();
         assertEquals(connectedClients2.size(), 0);
     }
+
+    @Test
+    public void testConnectionRemovedOnce() {
```

Comment:
Can you provide a more descriptive name? The remove is meant for the listener, but it isn't obvious from the test name. testDestroyConnection_connectionRemovedCalledOnce